more filtering

ahnitz opened the pull request:
Add an explicity fir lowpass / highpass filter. Speed up application of the filtering by switching to an FFT approach. This duplicates the functionality of the scipy.lfilter function.

spxiwh commented on pycbc/filter/resample.py:
```diff
@@ -236,6 +296,44 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
+    frequency: float
+        The frequency below which is suppressed. 
+    order: int
+        Number of corrupted samples on each side of the time series
```
Document the beta kwarg

spxiwh commented on pycbc/filter/resample.py:
```diff
@@ -236,6 +296,44 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
+    frequency: float
+        The frequency below which is suppressed. 
+    order: int
+        Number of corrupted samples on each side of the time series
```
Document the beta kwarg

spxiwh commented:
Hi Alex,

In principle this looks good to me, but is there some sanity check, or test in the test directory, that can be run to verify that resample_to_delta_t still works nicely?

Cheers
Ian

ahnitz commented:
Ian, I've updated the docstrings, and added a unittest for lfilter that runs in travis. It just verifies that using random input you get the same result as scipy would give.

ahnitz commented:
https://www.atlas.aei.uni-hannover.de/~ahnitz/LSC/ahnitz/projects/code_testing/q1/filterx/local-site-scratch/html/

ahnitz commented:
Looks like some fluctuation in the chisq of short templates. https://www.atlas.aei.uni-hannover.de/~ahnitz/LSC/ahnitz/projects/code_testing/romopt/t2/output/local-site-scratch/html/3._single_triggers/3.05_H1_loudest_triggers/
Compare to without this patch.
https://www.atlas.aei.uni-hannover.de/~ahnitz/LSC/ahnitz/projects/code_testing/q1/filterx/local-site-scratch/html/3._single_triggers/3.05_H1_loudest_triggers/

The change to the chisq is simply to truncate the bins when the waveform says it has reached the stop frequency.

spxiwh commented:
Merged this now. But with "old-school" integer division (ie. 3 / 2 = 1), can we explicitly use the integer division operator 3 // 2 = 1? Otherwise you have to check to see if __future__ division was imported anywhere to see what it is doing. To me 3 / 2 = 1.5 ... always.

ahnitz commented:
I don't really agree with you on this one Ian. To me integer division is
integer division.

On Tue, Mar 29, 2016 at 11:49 AM, spxiwh <notifications@github.com> wrote:

> Merged this now. But with "old-school" integer division (ie. 3 / 2 = 1),
> can we explicitly use the integer division operator 3 // 2 = 1? Otherwise
> you have to check to see if *future* division was imported anywhere to
> see what it is doing. To me 3 / 2 = 1.5 ... always.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ligo-cbc/pycbc/pull/731#issuecomment-202810084>
>



-- 
Dr. Alexander Nitz
Max Planck Institute for Gravitational Physics (Albert Einstein Institute)
Callinstrasse 38
D-30167 Hannover,  Germany
Tel: +49 511 762-17097

spxiwh commented:
Reading the python developers thoughts on this:

https://www.python.org/dev/peps/pep-0238/

suggests to me a strong preference for using a // b ... especially as 3/2 = 1.5 in python 3.

spxiwh commented:
(and it makes sense to make our code python 3-proof where it is easy to do so)
